storage_v2/factoryadapter/reader.go

```diff
@@ -40,27 +40,36 @@ func NewTraceReader(spanReader spanstore.Reader) *TraceReader {
 	}
 }
 
-func (tr *TraceReader) GetTrace(ctx context.Context, traceID pcommon.TraceID) iter.Seq2[ptrace.Traces, error] {
-	return func(yield func(ptrace.Traces, error) bool) {
-		t, err := tr.spanReader.GetTrace(ctx, model.TraceIDFromOTEL(traceID))
-		if err != nil {
-			// if not found, return an empty iterator, otherwire yield the error.
-			if !errors.Is(err, spanstore.ErrTraceNotFound) {
-				yield(ptrace.NewTraces(), err)
+func (tr *TraceReader) GetTraces(
+	ctx context.Context,
+	traceIDs ...tracestore.GetTraceParams,
+) iter.Seq2[[]ptrace.Traces, error] {
+	return func(yield func([]ptrace.Traces, error) bool) {
+		for _, idParams := range traceIDs {
+			// TODO start/end times are not supported by v1 reader
+			// https://github.com/jaegertracing/jaeger/pull/6242
+			t, err := tr.spanReader.GetTrace(ctx, model.TraceIDFromOTEL(idParams.TraceID))
+			if err != nil && !errors.Is(err, spanstore.ErrTraceNotFound) {
```

Comment:
do we have a test for the `ErrTraceNotFound` case?

Reply:
also - shouldn't we continue if the trace isn't found?

Reply:
good catch